### DISCUSSION TITLE
[BUG] SVG rendering failed on rofi version 1.6.1

### DISCUSSION THREAD

**Discussion by ztlevi** [davatorium/rofi#1235] [Distro Specific]
SVG rendering failed on rofi version 1.6.1. Fallback the version to 1.5.4 and rendering works normally.

## Version

Version: 1.6.1

## Configuration

- rofi -help
https://gist.githubusercontent.com/ztlevi/6cb8d327e3cf64b40b80443db84fb450/raw/53655a9321862199293eafd93051fb637a493b64/gistfile1.txt

## Launch Command

```
rofi -terminal alacritty -m -1 \
  -show drun \
  -modi drun,run \
  -show-icons \
  -theme theme/appmenu.rasi
```

## Steps to reproduce

- Install flat-remix-icon-theme
- Install latest 1.6.1 version rofi
- Start rofi 

## What behaviour you see

Error message like this. I tried Papirus icon theme here but it actually break for any svg based icons.
```
(process:6549): Helpers.IconFetcher-WARNING **: 02:21:37.073: Failed to load image: Couldn’t recognize the image file format for file “/etc/profiles/per-user/ztlevi/share/icons/Papirus/24x24/categories/google-chrome.svg”

(process:6549): Helpers.IconFetcher-WARNING **: 02:21:37.073: Failed to load image: Couldn’t recognize the image file format for file “/etc/profiles/per-user/ztlevi/share/icons/Papirus/24x24/categories/calc.svg”
```

## What behaviour you expect to see

Svg icons can rendered normally.

**Additional details:**
- Include a link to your private config: https://github.com/ztlevi/nix-dotfiles/tree/flake/modules/themes/rainbow/config/rofi
- Include screenshots/casts of your issue

**Comment by DaveDavenport** [comment]
1.6.1 uses libgdk-pixbuf to open images (to get support for a lot more images).
It opens svg images fine for me, you might want to check your gdk-pixbuf installation.

You can get extra debug output with `G_MESSAGES_DEBUG=Helpers.IconFetcher rofi -show drun`  to see what formats gdk-pixbuf reports to support.

**Comment by ztlevi** [comment]
using `G_MESSAGES_DEBUG=Helpers.IconFetcher rofi -show drun` reports:
```
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: ani
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: bmp
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: gif
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: icns
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: ico
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: cur
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: jpeg
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: jpe
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: jpg
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: png
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: pnm
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: pbm
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: pgm
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: ppm
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: qtif
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: qif
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: tga
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: targa
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: tiff
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: tif
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: xbm
Helpers.IconFetcher-INFO: 17:44:26.562: Add image extension: xpm
```

No svg listed. 

Since I'm using nixos, I double-checked the `pkg-pixbuf` version which is 2.42, https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/gdk-pixbuf/default.nix#L28.

I tried to install `pkg-pixbuf` manually and also tried to build rofi 1.6.1 from source with gdk-pixbuf, it gives me same reports which doesn't have svg support.

**Comment by lbonn** [comment]
@ztlevi svg support clearly seems to be missing from the nixos version, it should probably be reported as an issue in nixpkgs.

For comparison, here is what guix does https://git.savannah.gnu.org/cgit/guix.git/tree/gnu/packages/gtk.scm#n604

**Comment by DaveDavenport** [comment]
Yep, this seems to be a NixOs issue.  Reminds me of the old gentoo days, gmpc users with broken gdk-pixbufs was daily fun.

**Comment by sardemff7** [comment]
FTR, `librsvg` installs a `gdb-pixbuf` plugin that is loaded at runtime. But that module needs to be registered at install time, [which seems to be the case in the nixos package too](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/librsvg/default.nix#L69). However, there is no dependency on `librsvg` from `gdk-pixbuf`, so you can just *not* install `librsvg` and never notice anything before trying to load an SVG.

**Comment by ztlevi** [comment]
@sardemff7 Thanks for reply. But I'm confused... Shouldn't this line configure gdk-pixbuf already? https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/librsvg/default.nix#L52

**Comment by sardemff7** [comment]
The `postInstall` part creates a *full* file in `librsvg` directory/whatever it’s called, but the `postInstall` part is merging it to the “main” file, AFAIU.

**Comment by DaveDavenport** [comment]
Testing discussions. Lets see how this works.

**Comment by GG2R10** [comment]
Had the same problem on version 1.7.8-2 after a system update in Arch. The issue was a missing file of the library libgdk_pixbuf-2.0.so.0. Solved after reinstalling the gdk_pixbuf2 package with --overwrite '*' :D
